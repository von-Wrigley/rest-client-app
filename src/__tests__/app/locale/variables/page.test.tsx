import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import VariablesPage from "@/app/[locale]/variables/page";

jest.mock("../../../../components/shared/auth-guard", () => ({
  __esModule: true,
  default: ({ children }: { children: React.ReactNode }) => (
    <div data-testid="auth-guard">{children}</div>
  ),
}));

jest.mock("../../../../components/variables", () => ({
  __esModule: true,
  default: () => (
    <div data-testid="variables-component">Variables Component</div>
  ),
}));

describe("VariablesPage", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("renders without crashing", async () => {
    render(await VariablesPage());
    expect(screen.getByTestId("auth-guard")).toBeInTheDocument();
  });

  it("wraps the Variables component with AuthGuard", async () => {
    render(await VariablesPage());

    const authGuard = screen.getByTestId("auth-guard");
    const variablesComponent = screen.getByTestId("variables-component");

    expect(authGuard).toContainElement(variablesComponent);
  });

  it("has proper suspense boundaries structure", async () => {
    const { container } = render(await VariablesPage());
    const outerSuspense = container.firstChild;
    expect(outerSuspense).toBeInTheDocument();
    const authGuard = screen.getByTestId("auth-guard");
    expect(authGuard).toBeInTheDocument();
    const variablesComponent = screen.getByTestId("variables-component");
    expect(variablesComponent).toBeInTheDocument();
  });

  it("matches snapshot", async () => {
    const { asFragment } = render(await VariablesPage());
    expect(asFragment()).toMatchSnapshot();
  });
});
