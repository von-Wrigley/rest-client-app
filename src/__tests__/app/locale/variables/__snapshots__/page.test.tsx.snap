// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VariablesPage matches snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="auth-guard"
  >
    <div
      data-testid="variables-component"
    >
      Variables Component
    </div>
  </div>
</DocumentFragment>
`;
